lib/fauxpaas/cli/main.rb

```diff
@@ -7,15 +7,33 @@ module CLI
     class Main < Thor
 
       option :branch,
-          type: :string,
-          default: "master",
-          desc: "The branch or revision to deploy"
+        type: :string,
+        aliases: "-b",
+        desc: "The branch to deploy. " +
+          "Use default_branch to display or set the default branch."
+
+      option :revision,
+        type: :string,
+        aliases: "-r",
+        desc: "The revision to deploy. If unspecified, the latest is deployed."
 
       desc "deploy <instance>",
-      "Deploys the instance's source; by default deploys master. Use --branch to deploy a specific revision"
+        "Deploys the instance."
       def deploy(instance_name)
         instance = Fauxpaas.instance_repo.find(instance_name)
-        Fauxpaas.deployer.deploy(instance,branch: :branch)
+        Fauxpaas.deployer.deploy(instance, branch: options[:branch], revision: options[:revision])
+      end
+
+      desc "default_branch <instance> [<new_branch>]",
+        "Display or set the default branch for the instance"
+      def default_branch(instance_name, new_branch = nil)
+        instance = Fauxpaas.instance_repo.find(instance_name)
+        if new_branch
+          instance.default_branch = new_branch
+          Fauxpaas.instance_repo.save(instance)
```

Comment:
The output is unclear from the callsite. I'd probably do a bare if and return after.

Reply:
I'm willfully ignoring what `.save` returns. If if-else branches should have different output messages, they should be prepared here. If it should always be `instance.default_branch`, it should happen after the no-else if.

Reply:
I don't get your meaning.

Reply:
```
if new_branch
  instance.default_branch = new_branch
  Fauxpaas.instance_repo.save(instance)
end

instance.default_branch
```
or

```
if new_branch
  instance.default_branch = new_branch
  Fauxpass.instance_repo.save(instance)
  "Some successful update message"
else
  "Message containing unchanged value"
end
```